### PR TITLE
Decompiled 7 functions from object_functions

### PR DESCRIPTION
**Pull request by DavidSM64**
Passed the 21% mark.

Decompiled: `obj_init_animation`, `obj_init_bombexplosion`, `obj_init_audio`, `obj_init_midifadepoint`, `obj_loop_posarrow`, `obj_loop_snowball`, `obj_loop_coincreator`

Renamed
* `unknown_00BC20` -> `objects`
* `func_8000EA54` -> `spawn_object`

### CODE REVIEW

File: src/object_functions.c [hunk 1]
```diff
@@ -1043,7 +1029,24 @@ void obj_init_posarrow(Object *obj, LevelObjectEntry_PosArrow *entry) {
...
+    Object *someObj;
+    s32 numberOfObjects;
+
+    obj->unk6 |= 0x4000;
+    someObjList = func_8001BAAC(&numberOfObjects);
+    if (obj->unk78 < numberOfObjects) {
+        someObj = someObjList[obj->unk78];
+        someObj64 = someObj->unk64;
+        if (someObj64->unk0 == -1) {
+            obj->unk6 &= 0xBFFF;
```
Comment (AntonioCastelli) [inline src/object_functions.c:1044]:
```suggestion
            obj->unk6 &= ~0x4000;
```
IDK what this represents, but I'd guess this is the intent.

File: src/object_functions.c [hunk 2]
```diff
@@ -1072,8 +1075,168 @@ void obj_init_animator(Object *obj, LevelObjectEntry_Animator *entry, s32 arg2)
...
+    segmentId = obj64->segmentId;
+
+    if (obj64->segmentId != -1) {
+        levelModel = func_8002C7C4();
+        levelModelSegment = &levelModel->segments[segmentId];
+        batchId = obj64->batchId;
+        triangleBatchInfo = &levelModelSegment->batches[batchId];
+        texIndex = triangleBatchInfo->textureIndex;
+        curFacesOffset = triangleBatchInfo->facesOffset;
+        nextFacesOffset = (triangleBatchInfo + 1)->facesOffset;
```
Comment (AntonioCastelli) [inline src/object_functions.c:1132]:
```suggestion
        nextFacesOffset = triangleBatchInfo[1].facesOffset;
```

File: src/object_functions.c [hunk 3]
```diff
@@ -2285,10 +2539,134 @@ void obj_init_wavegenerator(Object *obj, LevelObjectEntry_WaveGenerator *entry,
...
+  f32 y;
+  f32 z;
+  f32 dist;
+
+  obj64 = (LevelObjectEntry_MidiFadePoint *) obj->unk64;
+  obj64->unk2 = entry->unkA;
+  obj64->unk0 = entry->unk8;
+  obj64->unk1C = entry->unk1C;
+  if (obj64->unk2 < obj64->unk0)
+  {
```
Comment (AntonioCastelli) [inline src/object_functions.c:2648]:
```suggestion
  if (obj64->unk2 < obj64->unk0) {
```

File: src/object_functions.h [hunk 4]
```diff
@@ -450,6 +457,69 @@ typedef struct Object_64_Trigger {
...
+typedef struct Object_64_Audio {
+    /* 0x00 */ u16 unk0;
+    /* 0x02 */ u16 unk2;
+    /* 0x02 */ u8 unk4;
+    /* 0x02 */ u8 unk5;
+    /* 0x02 */ u8 unk6;
+    /* 0x02 */ u8 unk7;
+    /* 0x02 */ s32 unk8;
+    /* 0x02 */ u8 unkC;
+    /* 0x02 */ u8 unkD;
```
Comment (AntonioCastelli) [inline src/object_functions.h:471]:
```suggestion
    /* 0x00 */ u16 unk0;
    /* 0x02 */ u16 unk2;
    /* 0x04 */ u8 unk4;
    /* 0x05 */ u8 unk5;
    /* 0x06 */ u8 unk6;
    /* 0x07 */ u8 unk7;
    /* 0x08 */ s32 unk8;
    /* 0x0C */ u8 unkC;
    /* 0x0D */ u8 unkD;
```

File: src/objects.h [hunk 5]
```diff
@@ -182,7 +182,7 @@ unknown8011AECC *func_8001BA1C(s32 arg0, s32 arg1);
 s32 func_8001BA64();
```
Comment (AntonioCastelli) [inline src/objects.h:182]: Also update #include guard.

**Review by AntonioCastelli — COMMENTED** [review]
It's only tangential to this PR, but thoughts on also having these renamed?
`obj_init_coincreator` -> `obj_init_bananacreator`
`obj_loop_coincreator` -> `obj_loop_bananacreator`
`obj_init_coin` -> `obj_init_banana`
`obj_loop_coin` -> `obj_loop_banana`

**Review by Ryan-Myers — APPROVED** [review]
Nice work!

**Pull request merged**